web/packages/extension/src/common.ts

```diff
@@ -32,14 +32,62 @@ class CheckboxOption implements OptionElement<boolean> {
     }
 }
 
+class NumberInputOption implements OptionElement<number | null> {
+    constructor(
+        private readonly numberInput: HTMLInputElement,
+        readonly label: HTMLLabelElement
+    ) {
+        this.numberInput.addEventListener("input", () => {
+            this.numberInput.reportValidity();
+        });
+    }
+
+    get input() {
+        return this.numberInput;
+    }
+
+    get value(): number | null {
+        const ni = this.numberInput;
+        const num = ni.valueAsNumber;
+        if (Number.isNaN(num)) {
+            return null;
+        }
+        if (ni.min) {
+            const min = Number.parseFloat(ni.min);
+            if (min > num) {
+                return min;
+            }
+        }
+        if (ni.max) {
+            const max = Number.parseFloat(ni.max);
+            if (num > max) {
+                return max;
+            }
+        }
+        return num;
+    }
+
+    set value(value: number | null) {
+        if (value === null) {
```

Comment:
I find using the `?` and `??` operators here much more readable:
```ts
this.numberInput.value = value?.toString() ?? "";
```

Reply:
I would like to avoid `NaN` here, since https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number claims that `NaN` is not a "valid floating point number", and that the `<input type="number"/>` tag required a "valid floating point number" for `value` to be considered valid:

> A number input is considered valid when empty and when a single number is entered, but is otherwise invalid. If the [required](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#required) attribute is used, the input is no longer considered valid when empty.
>
> Note: Any number is an acceptable value, as long as it is a [valid floating point number](https://html.spec.whatwg.org/multipage/infrastructure.html#valid-floating-point-number) (that is, not [NaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN) or [Infinity](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Infinity)).

I am trying to use:

`this.numberInput.value = value === null ? "" : value.toString();`

instead to make that code more concise, though I have not yet tested it, I will test it before pushing.

Reply:
Got it. In this case the code can be further reduced to:
```ts
this.numberInput.value = value?.toString() ?? "";
```
What do you think?